atuin-client/src/database.rs

```diff
@@ -458,6 +469,36 @@ impl Database for Sqlite {
             }
         };
 
+        let (exit, exclude_exit, cwd, exclude_cwd, before, after) = match opt_filter {
+            Some(opts) => (
+                opts.exit,
+                opts.exclude_exit,
+                opts.cwd,
+                opts.exclude_cwd,
+                opts.before,
+                opts.after,
+            ),
+            None => (None, None, None, None, None, None),
+        };
```

Comment:
Haven't quite looked into what these queries are doing, but for this I'd suggest making the opt_filter param not optional and instead derive(Default) on the struct. It will default to all None internally which matches the logic you have here

Reply:
Something like this? 9597ad5faad8804720a0e23c618b537cd9942b3c